### PR TITLE
Patches screen coordinates

### PR DESCRIPTION
**Pull request by TjWhale**
I think this is enough to test drawing the patches on the screen using their screen coordinates and list of connections between them.

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
This has formatting issues that need fixing. (check the circleci format_script build details).

File: scripts/microbe_stage/patch_map_generator.as [hunk 1]
```diff
@@ -11,15 +11,26 @@ PatchMap@ runGeneration()
...
     }
+    patch0.setScreenCoordinates(Float2(100,100));
+
+    Patch@ patch1 = Patch("Pangonaian Mesopelagic", 1, getBiomeTemplate("mesopelagic"));
+    patch1.setScreenCoordinates(Float2(200,100));
+    patch1.addNeighbour(0);
+    patch0.addNeighbour(1);
+
+    Patch@ patch2 = Patch("Pangonian Epipelagic", 2, getBiomeTemplate("default"));
+    patch2.setScreenCoordinates(Float2(200,200));
```
Comment (hhyyrylainen) [inline scripts/microbe_stage/patch_map_generator.as]: I think the coordinates should be different for the different patches.

File: src/microbe_stage/patch.h [hunk 2]
```diff
@@ -83,11 +83,22 @@ class Patch : public Leviathan::ReferenceCounted {
...
+    {
+        return screenCoordinates;
+    }
+
     REFERENCE_COUNTED_PTR_TYPE(Patch);
 
 private:
     const int32_t patchId;
     std::string name;
+    Float2 screenCoordinates; //where the patch should be displayed in the gui.
```
Comment (hhyyrylainen) [inline src/microbe_stage/patch.h]: This comment should be like this to work with doxygen:
```
//! Where the patch should be displayed in the gui.
Float2 screenCoordinates;

File: scripts/microbe_stage/patch_map_generator.as (hunk 1 above)
Comment (TjWhale) [reply scripts/microbe_stage/patch_map_generator.as]: I thought I did put the patches in a triangle with different coordinates for each patch.

File: src/microbe_stage/patch.h (hunk 2 above)
Comment (TjWhale) [reply src/microbe_stage/patch.h]: Yeah ok I can change this tomorrow,

File: scripts/microbe_stage/patch_map_generator.as (hunk 1 above)
Comment (hhyyrylainen) [reply scripts/microbe_stage/patch_map_generator.as]: You are correct. I misread one of the numbers.
